src/map.cpp

```diff
@@ -457,297 +461,130 @@ const vehicle *map::vehproceed()
+    if( ter_turn_cost >= veh.of_turn ) {
+        veh.of_turn_carry = veh.of_turn;
+        veh.of_turn = 0;
+        return &veh;
     }
 
-    veh->of_turn -= ter_turn_cost;
-
-    // if not enough wheels, mess up the ground a bit.
-    if (!veh->valid_wheel_config()) {
-        veh->velocity += veh->velocity < 0 ? 2000 : -2000;
-        for (auto &p : veh->parts) {
-            const tripoint pp = pt + p.precalc[0];
-            const ter_id &pter = ter(pp);
-            if (pter == t_dirt || pter == t_grass) {
-                ter_set(pp, t_dirtmound);
-            }
-        }
-    }
+    veh.of_turn -= ter_turn_cost;
 
-    if (veh->skidding) {
-        if (one_in(4)) { // might turn uncontrollably while skidding
-            veh->turn (one_in(2) ? -15 : 15);
+    if( veh.skidding ) {
+        if( one_in( 4 ) ) { // might turn uncontrollably while skidding
+            veh.turn( one_in( 2 ) ? -15 : 15 );
         }
-    }
-    else if( pl_ctrl && rng(0, 4) > g->u.skillLevel("driving") && one_in(20) ) {
-        add_msg(m_warning, _("You fumble with the %s's controls."), veh->name.c_str());
-        veh->turn (one_in(2) ? -15 : 15);
+    } else if( pl_ctrl && rng(0, 4) > g->u.skillLevel("driving") && one_in(20) ) {
+        add_msg( m_warning, _("You fumble with the %s's controls."), veh.name.c_str() );
+        veh.turn( one_in( 2 ) ? -15 : 15 );
     }
     // Eventually send it skidding if no control
     // But not if it's remotely controlled
-    if( !pl_ctrl && !veh->boarded_parts().size() && one_in(10) ) {
-        veh->skidding = true;
+    if( !pl_ctrl && veh.boarded_parts().empty() && one_in(10) ) {
+        veh.skidding = true;
     }
+
+    // Where do we go
     tileray mdir; // the direction we're moving
-    if (veh->skidding) { // if skidding, it's the move vector
-        mdir = veh->move;
-    } else if (veh->turn_dir != veh->face.dir()) {
-        mdir.init (veh->turn_dir); // driver turned vehicle, get turn_dir
+    if( veh.skidding ) {
+        // if skidding, it's the move vector
+        mdir = veh.move;
+    } else if( veh.turn_dir != veh.face.dir() ) {
+        // driver turned vehicle, get turn_dir
+        mdir.init( veh.turn_dir ); 
     } else {
-      mdir = veh->face;          // not turning, keep face.dir
+        // not turning, keep face.dir
+        mdir = veh.face;
     }
-    mdir.advance (veh->velocity < 0? -1 : 1);
-    const int dx = mdir.dx();           // where do we go
-    const int dy = mdir.dy();           // where do we go
-    bool can_move = true;
-    // calculate parts' mount points @ next turn (put them into precalc[1])
-    veh->precalc_mounts(1, veh->skidding ? veh->turn_dir : mdir.dir());
 
-    int dmg_1 = 0;
+    if( veh.velocity != 0 ) {
+        mdir.advance( veh.velocity < 0 ? -1 : 1 );
+    }
 
-    std::vector<veh_collision> veh_veh_colls;
-    std::vector<veh_collision> veh_misc_colls;
+    tripoint dp;
+    dp.x = mdir.dx();
+    dp.y = mdir.dy();
```

Comment:
`dp.z` is not set and remains at 0, will this cause problems if the vehicle is actually on a different z-level?